### PR TITLE
Add first css control points to svg when path selected

### PR DESCRIPTION
**Pull request by Paula-Kli**
ToDo:

- [x] Check why for cat y-axis is switched

### CODE REVIEW

File: lively.morphic/svg-morph/svg-morph.js [hunk 1]
```diff
@@ -162,6 +164,63 @@ export class SVGMorph extends Morph {
...
+
+          this.svgPath.parentNode.appendChild(defaultPoint);
+      }
+      let lastElement = element;
+    }
+  }
+
+  createControlPointAt (id, x, y, color) {
+    let point = document.createElement('div');
+    point.className = 'control-point';
```
Comment (T4rikA) [inline lively.morphic/svg-morph/svg-morph.js:204]: set attributes via {}

File: lively.morphic/svg-morph/svg-morph.js [hunk 2]
```diff
@@ -162,6 +164,63 @@ export class SVGMorph extends Morph {
...
+    point.style.background = color;
+    point.style.position = 'absolute';
+    point.style.left = y + 'px';
+    point.style.top = x + 'px';
+    // point.style.background-radius = "25px"
+
+    return point;
+  }
+
+  removeAllControlPoints () {
```
Comment (T4rikA) [inline lively.morphic/svg-morph/svg-morph.js:217]: cant we use something like: `$('control-point.my-path-selection').remove();`

File: lively.morphic/svg-morph/svg-morph.js (hunk 1 above)
Comment (T4rikA) [inline lively.morphic/svg-morph/svg-morph.js:204]: Why did you add an extra `className`, maybe use `control-point.my-path-selection`?

File: lively.morphic/svg-morph/svg-morph.js [hunk 3]
```diff
@@ -162,6 +164,63 @@ export class SVGMorph extends Morph {
 
   get isSVGMorph () { return true; }
 
+  getControlPoints () {
+    this.removeAllControlPoints();
```
Comment (T4rikA) [inline lively.morphic/svg-morph/svg-morph.js:168]: just remove if there was a previous path

File: lively.morphic/svg-morph/svg-morph.js [hunk 4]
```diff
@@ -137,9 +138,10 @@ export class SVGMorph extends Morph {
     const tar = SVG(target);
     let selection_node;
     switch (tar.type) {
-      // case 'path':
-      //  selection_node = SVG(target.outerHTML);
-      //  break;
+      case 'path':
+        console.log('here');
+        selection_node = SVG(target.outerHTML);
+        this.getControlPoints();
```
Comment (T4rikA) [inline lively.morphic/svg-morph/svg-morph.js:144]: please get the control points based on the `selection_node`

**Review by T4rikA — CHANGES_REQUESTED** [review]
image
An example of a selection in inkscape

File: lively.morphic/svg-morph/svg-morph.js (hunk 3 above)
Comment (Paula-Kli) [reply lively.morphic/svg-morph/svg-morph.js:167]: if you have a look at the method you can see that I remove all ControlPoints that I can find and if I don't find any I don't remove any -> we don't get an error message

File: lively.morphic/svg-morph/svg-morph.js (hunk 2 above)
Comment (Paula-Kli) [reply lively.morphic/svg-morph/svg-morph.js:203]: where do we get the jQuery from?

Comment (Paula-Kli) [reply lively.morphic/svg-morph/svg-morph.js:193]: this way it also works even though we might have only 1 or no controlPoints and we don't get a collection

**Comment by Paula-Kli** [comment]
==== NEWS ====
image
Moving the Control Points now also changes the SVG

File: lively.morphic/svg-morph/svg-morph.js [hunk 5]
```diff
@@ -140,15 +152,107 @@ export class SVGMorph extends Morph {
...
 
+  getControlPoints () {
+    this.removeAllControlPoints();
+
+    const tar = SVG(this.target);
+    const targetPath = SVG(this.target).array();
+
+    for (let i = 0; i < targetPath.length; i++) {
+      let element = targetPath[i];
+      switch (element[0]) {
```
Comment (T4rikA) [inline lively.morphic/svg-morph/svg-morph.js]: Why a switch case here?

File: lively.halos/morph.js [hunk 6]
```diff
@@ -2117,6 +2117,10 @@ export default class Halo extends Morph {
     this.target.onHaloDoubleMouseDown(evt);
   }
 
+  onDoubleMouseDown (evt) {
```
Comment (T4rikA) [inline lively.halos/morph.js]: Where do we use this?

File: lively.morphic/svg-morph/svg-morph.js [hunk 7]
```diff
@@ -140,15 +152,107 @@ export class SVGMorph extends Morph {
...
+      }
+      let lastElement = element;
+    }
+  }
+
+  createControlPointAt (id, x, y, color) {
+    const t = SVG(this.svgPath);
+    let point = t.circle();
+
+    point.attr({
```
Comment (T4rikA) [inline lively.morphic/svg-morph/svg-morph.js:183]: Maybe we should think of a dict with default settings like the color. Not needed in this PR just an idea

File: lively.halos/morph.js (hunk 6 above)
Comment (Paula-Kli) [reply lively.halos/morph.js]: I thought it was something that happened when I rebased on your branch/ branched from your branch... it is nothing that I wrote at least that I remember

File: lively.morphic/svg-morph/svg-morph.js (hunk 5 above)
Comment (Paula-Kli) [reply lively.morphic/svg-morph/svg-morph.js]: this is old and would be a possibility to e.g. change color if we find certain line commands... but I can change it

File: lively.morphic/svg-morph/svg-morph.js (hunk 7 above)
Comment (Paula-Kli) [reply lively.morphic/svg-morph/svg-morph.js:183]: I think that is a good idea...
since I started creating points with different colors for different line commands (as stated above) it is just some "old code" which can easily be changed

File: lively.halos/morph.js (hunk 6 above)
Comment (T4rikA) [reply lively.halos/morph.js]: Hm shouldn´t be there, can you rebase again? Or just remove it here

File: lively.morphic/svg-morph/svg-morph.js (hunk 5 above)
Comment (T4rikA) [reply lively.morphic/svg-morph/svg-morph.js]: I don´t see a use case where we would need different colors for different lines, but maybe I am missing something

**Review by T4rikA — APPROVED** [review]
